pysrc/Control_EnvironmentGUI.py

```diff
@@ -290,10 +290,8 @@ def send_message(self, conn, message_type, data=""):
         :param data: Optional argument for passing data that may be needed in c++
         :param type: Any
         """
-        message = EnvironmentControl.create_message(self, message_type, data)
-        length = len(message)
-        length = length.to_bytes(4, "big")
-        message = str(length).encode("utf-8") + message.encode("utf-8")
+        jsonPart = EnvironmentControl.create_message(self, message_type, data)
+        message = len(jsonPart).to_bytes(4, "big") + jsonPart.encode("utf-8")
```

Comment:
instead of converting the starting bytes to a string, you have to leave them as raw bytes